packages/esm-form-entry-app/src/app/form-data-source/form-data-source.service.ts

```diff
@@ -86,6 +89,40 @@ export class FormDataSourceService {
     return datasource;
   }
 
+  private getMostRecentObsDataSource(formSchema: FormSchema) {
+    const conceptIdentifiers: Set<string> = new Set();
+    for (const page of formSchema.pages) {
+      for (const section of page.sections) {
+        this.extractMostRecentObsConceptIds(section.questions, conceptIdentifiers);
+      }
+    }
+
+    return (patient: string) => {
+      return this.fetchMostRecentObsValue(patient, [...conceptIdentifiers]);
+    };
+  }
+
+  private extractMostRecentObsConceptIds(questions: Array<Questions>, concepts: Set<string>) {
+    for (const question of questions) {
+      const useMostRecentValue = question.questionOptions?.useMostRecentValue ?? false;
+      if (useMostRecentValue === 'true' || (typeof useMostRecentValue === 'boolean' && useMostRecentValue)) {
```

Comment:
@ibacher is this going to be a string named `"true"` or a boolean `true`

Reply:
Or this is an angular syntax 🤔 ?

Reply:
The idea is to support both, since that's what is used elsewhere in the form engine.

Reply:
okay, thanks for the polish, it looks much better and makes a lot of sense now 👍 .